Marlin/src/gcode/lcd/M414.cpp

```diff
@@ -0,0 +1,44 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2020 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "../../inc/MarlinConfig.h"
+
+#if HAS_MULTI_LANGUAGE
+
+#include "../gcode.h"
+#include "../../MarlinCore.h"
+#include "../../lcd/marlinui.h"
+
+/**
+ * M414: Set the language for the UI
+ *
+ * Parameters
+ *  S<index> : The language to select
+ */
+void GcodeSuite::M414() {
+
+  if (parser.seenval('S'))
+    ui.set_language(parser.value_byte());
+
+}
```

Comment:
I think this should range-check the input, to be sure that it is not setting to an index which does not exist.

On a related note, how would a serial-based UI learn about supported languages?

It would seem much more intuitive to accept language codes here, rather than an index, but I understand that might cause parsing complications.

Reply:
Invalid values are simply ignored. It's a very machine-specific setting since the LCD UI is tailored to individual machines. This G-code is only being included because Creality chose to include it in their firmware, and not because it's currently in demand. Vendors and users may choose to extend this initial basic implementation in future according to their needs.